sqlglot/optimizer/qualify_columns.py

```diff
@@ -465,6 +472,10 @@ def _qualify_outputs(scope: Scope):
     scope.expression.set("expressions", new_selections)
 
 
+def _get_source_node(scope: Scope, source_name: str) -> t.Optional[exp.Expression]:
+    return seq_get(scope.selected_sources.get(source_name) or [], 0)
```

Comment:
why was this added?

Reply:
Reduce repetition - this piece of logic is used in 4 places where we want to map a source name back to its node.

Reply:
I'll try to see if I can refactor and get rid of this altogether, I don't like this addition either.